costmap_2d/include/costmap_2d/layer.h

```diff
@@ -116,6 +116,18 @@ class Layer
     return false;
   }
 
+  virtual bool isStaticLayer() {
+    return false;
+  }
+
+  virtual std::shared_ptr<std::vector<double>> getDistancesFromStaticMap() {
+    return std::shared_ptr<std::vector<double>>();
```

Comment:
Is that what you want? That means all callers get a pointer to an 'empty' object that could have been modified by anyone else that called earlier (and from any instance of the costmap class). The code as is will most likely be elided and optimized into a simple default constructor call of the caller's variable (as opposed to a copy constructor having to be called).